src/pages/Main/Main.js

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState, useRef } from 'react';
+import './main.scss';
+import Slide from './Slide/Slide';
+
+function Main() {
+  const [imgId, setImgId] = useState(1);
+  const interval = useRef();
+
+  const onleftClick1 = () => {
+    clearInterval(interval.current);
+    imgId === 1 ? setImgId(imageData.length) : setImgId(prevId => prevId - 1);
+  };
+
+  const onrightClick1 = () => {
+    clearInterval(interval.current);
+    imgId === imageData.length ? setImgId(1) : setImgId(prevId => prevId + 1);
+  };
+
+  useEffect(() => {
+    const cycleImage = () => {
+      imgId === imageData.length ? setImgId(1) : setImgId(prevId => prevId + 1);
+    };
+
+    interval.current = setInterval(cycleImage, 4000);
+    return () => {
+      clearInterval(interval.current);
+    };
+  }, [imgId]);
```

Comment:
```suggestion
  const [imgId, setImgId] = useState(1);

  const onleftClick1 = () => {
    imgId === 1 ? setImgId(imageData.length) : setImgId(prevId => prevId - 1);
  };

  const onrightClick1 = () => {
    imgId === imageData.length ? setImgId(1) : setImgId(prevId => prevId + 1);
  };

  useEffect(() => {
    const cycleImage = () => {
      imgId === imageData.length ? setImgId(1) : setImgId(prevId => prevId + 1);
    };

    const autoSlide = setInterval(cycleImage, 4000);
    return () => {
      clearInterval(autoSlide);
    };
  }, [imgId]);
```
해당 로직에서 useRef를 사용하지 않아도 충분히 동작합니다.
추가로 함수 이름 수정이 필요할 것 같습니다.
함수 이름은 유저가 하는 동작에 대한 이름이 아닌
실제 이 함수가 동작하는 이름으로 지어주는 것이 좋습니다.

예를 들어 현재는 onLeftClick 이라는 이름인데,
결국 왼쪽 방향키를 유저가 클릭하는 동작을 뜻하는데, 이러한 네이밍이 아닌
showPrevImage 와 같은 이름이 해당 함수가 하는 동작을 뜻하기에
이와 같은 이름을 짓는 것이 더 바람직합니다

Reply:
화살표 클릭 시 clearInterval 을 하지 않아도 되는 이유가 무엇인지 궁금합니다. 클릭 시 랜더링이 새로 시작되어서 그런것인가요?

Reply:
useEffect에서 의존성 배열을 [imgId]로 넣어줬는데,
결국 이 값이 변경 될 때마다 해당 side effect가 일어나게 되는 로직입니다.
다시 side effect가 일어나려 한다면 clear effect가 실행되어야 하기 때문에
useEffect 안에서 이미 처리가 되는 동작이라
함수 안에서 다시 처리해줄 필요가 없게됩니다!

Reply:
아하 이해했습니다!